src/shimoku_api_python/api/plot_api.py

```diff
@@ -7672,6 +7672,157 @@ def line_with_confidence_area(self,
             }
         )
 
+    @logging_before_and_after(logging_level=logger.info)
+    def tornado(
+            self,
+            menupath: str,
+            order: int,
+            df_neg: pd.DataFrame, df_pos: pd.DataFrame,
+            cols_size=12,
+            rows_size=5,
+            neg_col_val="value", pos_col_val="value",
+            neg_col_name="name", pos_col_name="name",
+            options_mod={}, neg_dataops_mod={},
+            pos_dataops_mod={}, seriesopt_mod={},
+    ):
+        """
+        Plot a Tornado chart
+        - options_mod: level 1 echarts options
+        https://echarts.apache.org/en/option.html
+        - for pos_dataops_mod and neg_dataops_mod, see
+        https://echarts.apache.org/en/option.html#series-bar.data
+        - for seriesopt_mod see
+        https://echarts.apache.org/en/option.html#series-bar.type
+        """
+        def deep_update(mapping: Dict, *updating_mappings: Dict) -> Dict:
+            """
+            Updates a nested dictionary
+            """
+            # https://github.com/pydantic/pydantic/blob/fd2991fe6a73819b48c906e3c3274e8e47d0f761/pydantic/utils.py#L200
+            updated_mapping = mapping.copy()
+            for updating_mapping in updating_mappings:
+                for k, v in updating_mapping.items():
+                    if k in updated_mapping and isinstance(updated_mapping[k], dict) and isinstance(v, dict):
+                        updated_mapping[k] = deep_update(updated_mapping[k], v)
+                    else:
+                        updated_mapping[k] = v
+            return updated_mapping
+
+        # Default echarts options for tornado
+        pos_dataops_org = {
+            'label': {
+                'position': 'right',
+            },
+            'itemStyle': {
+                'color': 'var(--chart-C2)',
+                'borderRadius': [0, 9, 9, 0],
+            },
+            'emphasis': {
+                'itemStyle': {
+                    'color': 'var(--chart-C2)',
+                    'borderColor': 'var(--chart-C2)',
+                }
+            },
+        }
+
+        neg_dataops_org = {
+            'label': {
+                'position': 'left',
+            },
+            'itemStyle': {
+                'color': 'var(--complementary-red-light)',
+                'borderRadius': [9, 0, 0, 9],
+            },
+            'emphasis': {
+                'itemStyle': {
+                    'color': 'var(--complementary-red-light)',
+                    'borderColor': 'var(--complementary-red-light)',
+                }
+            },
+        }
+
+        options_org = {
+            'toolbox': {
```

Comment:
This could use the default options of the toolbox, the magic type can be removed, see the scatter_with_effect for example